lib/rspec-puppet/support.rb

```diff
@@ -279,6 +279,34 @@ def trusted_facts_hash(node_name)
       extensions
     end
 
+    def server_facts_hash
+      server_facts = {}
+
+      # Add our server version to the fact list
+      server_facts["serverversion"] = Puppet.version.to_s
+
+      # And then add the server name and IP
+      {"servername" => "fqdn",
+        "serverip" => "ipaddress"
+      }.each do |var, fact|
+        if value = Facter.value(fact)
+          server_facts[var] = value
+        else
+          Puppet.warning _("Could not retrieve fact %{fact}") % { fact: fact }
```

Comment:
Please use `warn` here instead of `Puppet.warning`, this will prevent the message from requiring a properly configured Puppet logger.

Also, rspec-puppet doesn't currently externalise strings, so you'll want to get rid of the `_` method call to get the tests to pass.

Reply:
Ah, that's where those Travis failures came from... Will get this updated... 😀